apps/rush-lib/src/cli/actions/ListAction.ts

```diff
@@ -21,47 +29,73 @@ export class ListAction extends BaseRushAction {
     });
   }
 
+  public setJsonObject(name: string, version: string, path: string): IJsonEntity {
+    return {name: name, version: version, path: path};
+  }
+
   protected onDefineParameters(): void {
     this._version = this.defineFlagParameter({
       parameterLongName: '--version',
       parameterShortName: '-v',
-      description: 'If this flag is specified, the project version will be ' +
+      description:
+        'If this flag is specified, the project version will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._path = this.defineFlagParameter({
       parameterLongName: '--path',
       parameterShortName: '-p',
-      description: 'If this flag is specified, the project path will be ' +
+      description:
+        'If this flag is specified, the project path will be ' +
         'displayed in a column along with the package name.'
     });
 
     this._fullPath = this.defineFlagParameter({
       parameterLongName: '--full-path',
       parameterShortName: '-f',
-      description: 'If this flag is specified, the project full path will ' +
+      description:
+        'If this flag is specified, the project full path will ' +
         'be displayed in a column along with the package name.'
     });
+
+    this._jsonFlag = this.defineFlagParameter({
+      parameterLongName: '--json',
+      description: 'If specified, output will be in JSON format.'
+    });
   }
 
   protected run(): Promise<void> {
     return Promise.resolve().then(() => {
       const allPackages: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
       if (this._version.value || this._path.value || this._fullPath.value) {
         this._printListTable(allPackages);
+      } else if (this._jsonFlag.value) {
+        this._serializeToJson(allPackages);
       } else {
         this._printList(allPackages);
       }
     });
   }
 
+  private _serializeToJson(
```

Comment:
The name `_serializeToJson` implies that this function returns a JSON-serialized string. Let's rename this to something like `_printJson`

```suggestion
  private _printJson(
```